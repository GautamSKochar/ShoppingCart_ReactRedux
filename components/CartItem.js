import React from 'react';
import propTypes from 'prop-types';

class CartItem extends React.Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick() {
        this.props.handleClick(this.props.index);
    }

    render() {
        return (
            <div className="card" style={{ display: 'inline-block', margin: '25px' }}>
                <img className="card-img-top" src={this.props.product.imagePath} alt="Product" />
                <div className="caption">
                    <h3>{name}</h3>
                    <div className="card-text" style={{ display: 'inline', 'lineHeight': '34px' }}>
                        {this.props.product.name}
                    </div>
                    <div className="card-block" style={{ float: 'right', margin: '0px 5px 7px 5px' }}>
                        <button style={{ cursor: 'pointer' }} className="btn btn-danger" onClick={this.handleClick}>
                            Remove
                        </button>
                    </div>
                </div>
            </div>
        );
    }
}

CartItem.propTypes = {
    product: propTypes.isRequired,
    product: propTypes.object,
    index: propTypes.number
}

export default CartItem;