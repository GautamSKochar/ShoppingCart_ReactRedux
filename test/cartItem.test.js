import React from "react";
import { shallow, mount, render } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
import CartItem from "../components/CartItem";

const product = {
    "id": 1,
    "name": "BagPack",
    "price": 499,
    "currency": "Rs.",
    "imagePath": "images/backpack.jpg"
}
const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });
// const store = mockStore({});
describe('CartItem Component', () => {
    it('Snapshot Test', () => {
        const component = shallow(<CartItem index={product.id} product={product} />);
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    })
});