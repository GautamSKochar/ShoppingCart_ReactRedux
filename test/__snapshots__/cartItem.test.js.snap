// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component Snapshot Test 1`] = `
<div
  className="card"
  style={
    Object {
      "display": "inline-block",
      "margin": "25px",
    }
  }
>
  <img
    alt="Product"
    className="card-img-top"
    src="images/backpack.jpg"
  />
  <div
    className="caption"
  >
    <h3>
      nodejs
    </h3>
    <div
      className="card-text"
      style={
        Object {
          "display": "inline",
          "lineHeight": "34px",
        }
      }
    >
      BagPack
    </div>
    <div
      className="card-block"
      style={
        Object {
          "float": "right",
          "margin": "0px 5px 7px 5px",
        }
      }
    >
      <button
        className="btn btn-danger"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Remove
      </button>
    </div>
  </div>
</div>
`;
